\documentclass{scrartcl}

\usepackage[utf8]{inputenc}
\usepackage{hyperref}
\usepackage{cleveref}
\usepackage{url}
\usepackage{graphicx}
\usepackage{amssymb}
\usepackage[T1]{fontenc}
\usepackage{blkarray}
\usepackage{caption}
\usepackage{subcaption}
\usepackage{biblatex}
\addbibresource{bibliography.bib} 


\newcommand{\emailaddr}[1]{\href{mailto:#1}{\texttt{#1}}}


\title{EDAM
\\
Ontology of bioscientific data analysis and data management
\\
\begin{small} 
  Semantic Web - 
  MSc Computer Engineering and Science
\end{small}
}
\author{
    \emailaddr{davide.domini@studio.unibo.it}
}
\date{March 2023}

\begin{document}

\maketitle

\begin{abstract}
  The EDAM \footnote{http://edamontology.org/page} ontology is a comprehensive and structured resource for the classification and organization of bioinformatics data, 
    tools, and concepts. It encompasses over 2,000 terms across multiple domains, including data types, data formats, software tools, 
    and data analysis methods. The ontology is designed to facilitate the integration and sharing of bioinformatics resources and knowledge 
    across various disciplines, enabling researchers to more effectively collaborate and build upon one another's work.

  EDAM's structure is built around a set of hierarchical relationships between terms, allowing users to navigate between related concepts 
    and find relevant resources more easily. Additionally, EDAM provides a rich set of metadata for each term, including definitions, synonyms,
    and cross-references to related terms and resources.
  
  The ontology is regularly updated by a community of experts and is freely available for use and integration with other bioinformatics 
    resources. 
\end{abstract}


\newpage
\tableofcontents

\newpage
\listoffigures

\newpage

\section{Overview}

\subsection{Goals}

Bioinformatics is a rapidly growing field that involves analyzing biological data using computational 
  tools and methods. With the increasing number and diversity of bioinformatics tools and data resources, 
  there is a growing need for effective ways to organize, find, understand, compare, select, use, and connect 
  the available resources.

However, these tasks often rely on consistent, machine-readable descriptions of the underlying components, 
  which have been lacking in ad hoc resource descriptions. This has made it challenging for researchers to
  effectively use and integrate various bioinformatics tools and resources into their workflows or workbenches.

To address this challenge, a comprehensive ontology called EDAM \cite{edam} (EMBRACE Data And Methods) has been introduced. 
  EDAM unifies semantically the bioinformatics concepts in common use and provides a comprehensive controlled 
  vocabulary that is broadly applicable to various applications.

The primary goal of EDAM is to create coherent, machine-readable annotations for use within resource catalogues. 
  This ontology offers a standardized way to categorize bioinformatics tools and data resources based on a 
  number of categories, including their application domain (e.g. protein structure, metagenomics), 
  function (e.g. alignment construction), type of input and output data (e.g. accession, feature record), 
  and available formats of the data (e.g. FASTQ, PDB format).

EDAM supports new and powerful search, browse, and query functions, enabling users to efficiently find and 
  compare bioinformatics tools and resources. It is intended to complement standards for data exchange, 
  enrich provenance metadata, and offer a shared markup vocabulary for bioinformatics data on the Semantic Web.

Furthermore, EDAM is designed to aid text mining by defining interrelated terms and synonyms. 
  It is also intended to be conveniently usable by annotators and tool users ranging from programmers to 
  lab biologists.

In summary, EDAM offers a much-needed ontology that unifies bioinformatics concepts, provides a comprehensive 
  controlled vocabulary, and supports new and powerful search, browse, and query functions. It has the potential 
  to greatly facilitate the search, publication, and integration of bioinformatics tools and resources, 
  thereby advancing research in the field.


\subsection{Related Work}

\newpage

\section{EDAM Core}

EDAM includes 5 main sub-ontologies of concepts (\Cref{fig:concepts}):

\begin{figure}[h!]
  \centering
  \includegraphics[width=0.7\textwidth]{imgs/EDAMconcepts.png}
  \caption{EDAM concepts}
  \label{fig:concepts}
\end{figure}

\subsection{Structure}

\subsubsection{Concepts}
\begin{itemize}
  \item \textbf{Operation}: A function that processes a set of inputs and results in a set 
    of outputs, or associates arguments (inputs) with values (outputs). 
    Special cases are: 
      i) An operation that consumes no input (has no input arguments). Such operation is either 
      a constant function, or an operation depending only on the underlying state. 
      ii) An operation that may modify the underlying state but has no output. 
      iii) The singular-case operation with no input or output, that still may modify the underlying state.

  \item \textbf{Data}: Information, represented in an information artefact (data record) 
    that is ‘understandable’ by dedicated computational tools that can use 
    the data as input or produce it as output.

  \item \textbf{Identifier}: A text token, number or something else that identifies an entity, 
    but which may not be persistent (stable) or unique (the same identifier may identify multiple
    things).
  
  \item \textbf{Topic}: A category denoting a rather broad domain or field of interest, of study, 
    application, work, data or technology. Topics have no clearly defined borders between each other.
  
  \item \textbf{Format}: A defined way or layout of representing and structuring data in a computer file, 
    blob, string, message or elsewhere.
  
\end{itemize}

\subsubsection{Relations}

\begin{figure}[h!]
  \centering
  \includegraphics[width=0.7\textwidth]{imgs/EDAMrelations.png}
  \caption{EDAM relations}
  \label{fig:relations}
\end{figure}

\subsubsection{Concepts types} 

\subsection{How EDAM might be used}


\newpage

\section{Usecases}

\subsection{Use case 1}
\subsection{Use case 2}

%Bibliography
\newpage
\addcontentsline{toc}{section}{Bibliography}
\printbibliography %Prints bibliography


\end{document}